sechub-test/src/test/java/com/daimler/sechub/test/RoutesTest.java

```diff
@@ -0,0 +1,174 @@
+				if (route.contains("anonymous") || route.contains("actuator")) {
+					assertTrue(clazzName + " with route " + route + " should not contain roles.", roles.length == 0);
+				}
+				
+				if (route.contains("/api/admin")) {
+					assertTrue(clazzName + " with route " + route + " should only contain one role, ROLE_SUPERADMIN.", roles.length == 1);
+					assertTrue(clazzName + " with route " + route + " should contain ROLE_SUPERADMIN.", Arrays.asList(roles).contains(RoleConstants.ROLE_SUPERADMIN));
+				}
+				
+				if (route.contains("/api/project") || route.contains("/api/job")) {
+					assertTrue(clazzName + " with route " + route + " should contain ROLE_USER.", Arrays.asList(roles).contains(RoleConstants.ROLE_USER));
+					assertTrue(clazzName + " with route " + route + " should contain ROLE_SUPERADMIN.", Arrays.asList(roles).contains(RoleConstants.ROLE_SUPERADMIN));
+				}
+			}	
+		}
+	}
+	
+	private List<Class> getAllClassesForAnnotationType(Class clazz) {
+		ClassPathScanningCandidateComponentProvider scanner = new ClassPathScanningCandidateComponentProvider(false);
+		scanner.addIncludeFilter(new AnnotationTypeFilter(clazz));
+		
+		Set<BeanDefinition> allRestControllerBeans = scanner.findCandidateComponents("com.daimler.sechub");
+		
+		return allRestControllerBeans.stream()
+				.map( bean -> {
+					try {
+						return Class.forName(bean.getBeanClassName());
+					} catch (Exception e) {
+						e.printStackTrace();
+					}
+					return null;
+				})
+				.collect(Collectors.toList());
+	}
+	
+	private void getClassUserRolePaths(Class clazz) {
+		RequestMapping classMappings = AnnotationUtils.findAnnotation(clazz, RequestMapping.class);
+		
+		ArrayList<String> userRoles = new ArrayList<String>();
+		List<String> resultPaths = new ArrayList<String>();
+		
+		String basePath = "";
+		
+		if (classMappings != null) {
+			String[] paths = classMappings.path();
+			for (String path : paths) {
+				basePath = path;
+				
+				String[] classRoles = getRoles(clazz);
+				
+				if (classRoutesRolesMap.get(clazz) == null) {
+					classRoutesRolesMap.put(clazz, new HashMap<String, ArrayList<String>>());
+				}
+				if (classRoutesRolesMap.get(clazz).get(path) == null) {
+					classRoutesRolesMap.get(clazz).put(path, new ArrayList<String>());
+				}
+				
+				if (classRoles != null) {
+					userRoles.addAll(Arrays.asList(classRoles));
+				}
+				classRoutesRolesMap.get(clazz).get(path).addAll(userRoles);
+			}
+		}
+		
+		Method[] methods = clazz.getMethods();
+		
+		for (Method method : methods) {
+			RequestMapping methodMapping = AnnotationUtils.findAnnotation(method, RequestMapping.class);
+			if (methodMapping != null) {
+				String[] paths = methodMapping.path();
+				if (paths != null) {
+					for (String path : paths) {
+						
+						String fullPath = basePath + path;
+						
+						resultPaths.add(basePath + path);
+						
+						String[] methodRoles = getRoles(method);
+						
+						if (classRoutesRolesMap.get(clazz) == null) {
```

Comment:
With Java 8 you got a map.computIfAbsent method - maybe more readable when using this instead.